### PR TITLE
feat: claim rewards in the withdraw_lp_shares()

### PR DESCRIPTION
**Pull request by martinfridrich**
This PR fixes:
* #144
* #143

### CODE REVIEW

File: liquidity-mining/src/lib.rs [hunk 1]
```diff
@@ -1873,17 +1872,33 @@ impl<T: Config<I>, I: 'static> hydradx_traits::liquidity_mining::Mutate<T::Accou
         who: T::AccountId,
         deposit_id: DepositId,
         yield_farm_id: YieldFarmId,
-        fail_on_doubleclaim: bool,
     ) -> Result<(GlobalFarmId, T::AssetId, Self::Balance, Self::Balance), Self::Error> {
-        Self::claim_rewards(who, deposit_id, yield_farm_id, fail_on_doubleclaim)
+        const FAIL_ON_DOUBLECLAIM: bool = true;
```
Comment (enthusiastmartin) [inline liquidity-mining/src/lib.rs:1876]: i would put this somewhere else - at the top somewhere.

Is it actually needed at all? is there a scenario where you'd call claim_rewards with false ?

File: liquidity-mining/src/lib.rs [hunk 2]
```diff
@@ -1873,17 +1872,33 @@ impl<T: Config<I>, I: 'static> hydradx_traits::liquidity_mining::Mutate<T::Accou
...
         deposit_id: DepositId,
+        global_farm_id: GlobalFarmId,
         yield_farm_id: YieldFarmId,
-        unclaimable_rewards: Self::Balance,
-    ) -> Result<(GlobalFarmId, Self::Balance, bool), Self::Error> {
-        Self::withdraw_lp_shares(deposit_id, yield_farm_id, unclaimable_rewards)
+        amm_pool_id: Self::AmmPoolId,
+    ) -> Result<(Self::Balance, Option<(T::AssetId, Self::Balance, Self::Balance)>, bool), Self::Error> {
+        let claim_data = if Self::is_yield_farm_claimable(global_farm_id, yield_farm_id, amm_pool_id) {
+            const FAIL_ON_DOUBLECLAIM: bool = false;
```
Comment (enthusiastmartin) [inline liquidity-mining/src/lib.rs:1888]: same thing here.

File: traits/Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "hydradx-traits"
-version = "0.10.0"
+version = "1.0.0"
```
Comment (enthusiastmartin) [inline traits/Cargo.toml:3]: i'd leave at 0.11.0

**Review by enthusiastmartin — APPROVED** [review]
few minor comments

File: traits/Cargo.toml (hunk 3 above)
Comment (martinfridrich) [reply traits/Cargo.toml:3]: I changed api of LM's trait so major version should change.

**Pull request merged**